components/connector-service/internal/certificates/certificates.go

```diff
@@ -177,9 +177,17 @@ func (cu *certificateUtility) SignWithCA(caCrt *x509.Certificate, csr *x509.Cert
 		return "", apperrors.Internal("Error while creating certificate: %s", err)
 	}
 
-	clientCrt := &bytes.Buffer{}
+	serverCrt := pemEncodeCrt(caCrt.Raw)
 
-	pem.Encode(clientCrt, &pem.Block{Type: "CERTIFICATE", Bytes: clientCrtRaw})
+	clientCrt := pemEncodeCrt(clientCrtRaw)
 
-	return cu.encodeStringBase64(clientCrt.Bytes()), nil
+	certChain := append(clientCrt.Bytes(), serverCrt.Bytes()...)
```

Comment:
The good idea would be to extract encoding/appending to separate function that would accept two certificates, it could return whole chain as string/byte array. I am thinking about something like: 
`createCertChain(caCrt, clientCrt []byte) string`

You could also do the same treatment to the template creation in earlier part of this func:
`prepareCertificateTemplate(caCrt *x509.Certificate, csr *x509.CertificateRequest) x509.Certificate`

After this, the SignWithCA func would look something like this:
```
func (cu *certificateUtility) SignWithCA(caCrt *x509.Certificate, csr *x509.CertificateRequest, key *rsa.PrivateKey) (
	crtBase64 string, appError apperrors.AppError) {

	clientCRTTemplate := prepareCertificateTemplate(caCrt, csr)

	clientCrtRaw, err := x509.CreateCertificate(rand.Reader, &clientCRTTemplate, caCrt, csr.PublicKey, key)
	if err != nil {
		return "", apperrors.Internal("Error while creating certificate: %s", err)
	}

        certChain := createCertChain(caCrt.Raw, clientCrtRaw)
	
	return cu.encodeStringBase64(certChain), nil
}
```